### DISCUSSION TITLE
Missing python module netaddr

### DISCUSSION THREAD

**Discussion by bufanda** [semaphoreui/semaphore#1936] [Q&A] [answered]
Using the Jinja2 Filter 'ansible.utils.ipaddr' results in error with Message

```
"The ipaddr filter requires python's netaddr be installed on the ansible controller"
```

I am using the Docker Image  `semaphoreui/semaphore` in the latest version from docker hub.

This is a standard filter which comes with ansible and is installed per default 

https://docs.ansible.com/ansible/latest/collections/ansible/utils/docsite/filters_ipaddr.html

Shouldn't all default modules, collection and packages have their dependencies have installed per default too?

How can I solve this? Do I need to build the Docker Image myself and add all dependencies?
Couldn't find a solution to this problem or how to install custom python packages.

**Comment by fiftin** [comment]
Hi @bufanda it is external collection: https://galaxy.ansible.com/ui/repo/published/ansible/utils/

You can add it via `collections/requirements.yml`:

```yml
---
collections:
- ansible.utils
```

**Reply by tboerger** [reply]
That wouldn't resolve the issue the the python dependency for this collection module is missing.

**Reply by fiftin** [reply]
Aaa, got it.

**Reply by fiftin** [reply]
I see two ways:
1) create Docker image based on `semaphoreui/semaphore` and install required deps.
2) create Task Template which will install required deps:
```yml
- name: install netaddr
  pip:
    name: netaddr
```

**Reply by fiftin** [reply, marked as answer]
Just merged MR for installing python deps by requirements.txt:

https://github.com/semaphoreui/semaphore/pull/1250

**Reply by bufanda** [reply]
Is it though? The collection is installed per default with ansible in the base semaphoreui Image therefore the python dependecy should also be installed. I didn't add the collection in any of my requirements and it is available for use. If it  wouldn't be preinstallted I can understand your arguing but as it comes with the image in my humble opinion the image is incomplete and would need fixing and maintaining my own image of it is kinda a cheap cop out from your side in my opinion. 

Looks like I need to go back to AWX at least basics there work without me having to maintain my own version.

**Reply by fiftin** [reply]
@bufanda yes, of course you have to, no one will do your job for you.

**Reply by bufanda** [reply]
So then this software is useless when not even pre installed elements work correctly. Thanks for the clarification.

**Reply by fiftin** [reply]
@bufanda I didn’t think that this would have to be explained on GitHub, but this is open source software that is supported by the community. Instead of demanding something from someone, just implement it yourself. Pull requests are welcome.

**Reply by tboerger** [reply]
With the recent change you are able to provide an requirements.yml to the docker image which could install the netaddr dependency.

I'm currently building a refactoring of the docker image which embeds this dependency directly.

**Reply by johnsturgeon** [reply]
I'm currently looking at running python scripts and unfortunately there is no documentation.  I have installed the 'binary' (non docker) image and I was wondering how I can specify requirements to my python task runner?
